mdx2/io.py

```
@@ -100,3 +101,40 @@ def saveobj(obj, filename, name=None, append=False, mode="w"):
+
+        def _inspect_nexus_file(filename):
+            file_info = {}
+            mdx2_object_info = {}
+            try:
+                with h5py.File(filename, 'r') as f:
+                    file_info = {k:v for k, v in f.attrs.items()} # is this the best way to copy attrs into a dict?
+                    if 'mdx2_version' in file_info: # it's an mdx2 file, so look for mdx2 objects in the entry group
+                        for k, v in f['/entry'].items():
+                            mdx2_object_info[k] = {k:v for k, v in v.attrs.items()}
+                            mdx2_object_info[k]['keys'] = list(v.keys())
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)